script.js

```diff
@@ -203,12 +259,18 @@ function getTaskListMessage() {
     taskListMessageIconEl.className = "fa-solid fa-clipboard-check taskListMessageIcon";
 
     taskListMessageContainerEl.append(taskListMessageTextEl, taskListMessageIconEl)
-    // pane deleteTaskEl element taskToolsEl elemendi sisse
+    // pane taskListMessageContainerEl element taskListEl elemendi sisse
     taskListEl.appendChild(taskListMessageContainerEl);
 }
 
 // Kasutaja saab muuta ülesannet.
+function getModal() {
```

Comment:
Kustuta see ära. See kirjutab üle su eelmist getmodal funktsiooni

Reply:
ma olin juba, et waaaoow, kas töesti!? Aga see ei muutnud olukorda.

Valideeri, kas editTaskBtnEl.onclick töötab getModal funktsioonis.
  editTaskBtnEl.onclick = function () {
    console.log("clicking on edit task button");
  };

Küsin hoopis sellise küsimuse. Kui ma getEditBtnElement() sees panen editTaskIconEl klikk funktsiooni get modal avamisega, siis miks ma pean getModal() sees tegema klikk funktsiooni 



Jöudsin niipalju ära kirjutada ja siis hakkasin mötlema, aga miks see on seal?! Oota. Let me remove it - worked.

Reply:
sul getModal funktsiooni üldse ei käivitatagi, sest sul on 2x getModal kirjutatud - javascript kustutab esimese getModali üle teise getModaliga.

1. tee getModal funktsioonis `console.log('getting modal')` -> mitte midagi ei juhtud
2. kustuta getModal funktsioon ära realt 270 -> eelmine `console.log()` läheb tööle

Nüüd sa oled jõudnud kohta, kus tead, et edit nupul vajutamine töötab, getModal funktsioon käivitatakse ning saad hakata uurima, mis toimub getModal funktsiooni sees - miks see ei ava modalit.

Soovitan getModal ümber nimetada `showModalEl` funktsiooniks - see funktsioon ei tagasta sul modalit, nii et nimetus on vale. Kirjuta funktsioon ümber nii, et `showModalEl` teeks ainult ühte asja - näitaks modalit. 

Nüüd uuri, kes peaks käivitama `showModalEl` funktsiooni. 

---

Kui saad modali avanema, siis eraldi probleem on, kuidas modal sulgeda. Modal ei ole dünaamiline erinevalt tasklistist. Seega modali x-nupul võib kohe küljes olla mingi `closeModalEl` funktsioon. `<span onclick="closeModalEl()">Save</span>`


Ühtlasi, usun, et sul läheks lugemine selgemaks, kui alati kasutad samasugust stiili: otsusta, kas teed `el.onclick = () => {}` või `el.addEventlistener('click', () => {})`. Mulle tundub esimene lihtsamini loetav.

Reply:
okidoki! seal jah keerukus, kuidas teada, millist titlet muutma pead ja kuidas muudad.

Plaan on umbes selline, et modali avanedes pead käivitama funktsiooni, mis võtab endale kaasa konkreetse `taskTitle`'i, muudab `tasks` arrays õiget taski ning lõpuks käivitab `showAllTasks` funktsiooni uuendatud `tasks` arrayga.  `tasks` arrary muutmiseks soovitan kasutada vana head `.map` meetodit.

Reply:
Ma saan aru, et editTask funktsioon peab vötma sisse uue task inputi value.

Ja save buttonit vajutades ta peab käima läbi array, et leida see öige title, mida muuta tuleb uueks ja midal sulgeda.

Kui varasemalt vötsime elemendi attribute’i selleks, et leida see öige object, siis nüüd on nii mitu asja vahepeal, et kuidas see attribute saada? 🙄

Pushisin hetkel ära, kus ma kokku jooksin (katsetadin mitut moodi, aga ei saanud)…